### PR TITLE
{sql} Fix #19731: Add condition for SystemAssigned Identity type as well

### LINKED ISSUE #19731: az sql server update --identity-type SystemAssigned not working
## Describe the bug

Specifying `--identity-type SystemAssigned` together with `--assign_identity` results in the command being ignored and no managed identity is created.

Just using `--assign_identity` without `--identity-type SystemAssigned` works as expected.

Setting the default value for `--identity-type`  results in nothing happening.

**Command Name**
`az sql server update`

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- Create a sql server without any managed identity.
- `az sql server update --name {} --resource-group {} --assign_identity --identity-type SystemAssigned`

## Expected Behavior

`az sql server update --name {} --resource-group {} --assign_identity`

and 

`az sql server update --name {} --resource-group {} --assign_identity --identity-type SystemAssigned`

Should give the same result, which is that a System Assigned Managed Identity is created for the SQL Server.

## Environment Summary
```
Linux-5.4.0-1058-azure-x86_64-with-debian-10.2 (Cloud Shell), Common Base Linux Delridge (quinault)
Python 3.6.10
Installer: DEB

azure-cli 2.28.0

Extensions:
resource-graph 1.0.0
ai-examples 0.2.5
ssh 0.1.6

```

### PR DESCRIPTION
**Pull request by RakeshMohanMSFT**
Fix https://github.com/Azure/azure-cli/issues/19731

**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by strehan1993** [comment]
LGTM

**Comment by strehan1993** [comment]
LGTM

**Comment by yonzhan** [comment]
sql

File: src/azure-cli/azure/cli/command_modules/sql/custom.py [hunk 1]
```diff
@@ -497,6 +497,10 @@ def _get_identity_object_from_type(
 
                 identityResult = ResourceIdentity(type=ResourceIdType.user_assigned.value,
                                                   user_assigned_identities=umiDict)
+
+            if resourceIdentityType == ResourceIdType.system_assigned.value:
+                identityResult = ResourceIdentity(type=ResourceIdType.system_assigned.value)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/sql/custom.py:502]:
```suggestion
        if resourceIdentityType == ResourceIdType.system_assigned.value:
            identityResult = ResourceIdentity(type=ResourceIdType.system_assigned.value)
```
Do I read it wrong or is it really inside the if clause for `resourceIdentityType == ResourceIdType.user_assigned.value`?

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**